src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp

```diff
@@ -110,10 +111,87 @@ InitializeDampedHarmonic<Dim>::impl_rollon(
                          std::move(initial_d4_gauge_h));
 }
 
-template <size_t Dim>
+template <size_t Dim, bool UseRollon>
+tnsr::aa<DataVector, Dim, Frame::Inertial>
+InitializeDampedHarmonic<Dim, UseRollon>::new_pi_from_gauge_h(
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& spacetime_metric,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& pi,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+    const double sigma_r) noexcept {
+  const auto spatial_metric = gr::spatial_metric(spacetime_metric);
+  const auto det_and_inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric);
+  const auto& det_spatial_metric = det_and_inverse_spatial_metric.first;
+  const auto& inverse_spatial_metric = det_and_inverse_spatial_metric.second;
+  const Scalar<DataVector> sqrt_det_spatial_metric{
+      DataVector{sqrt(get(det_spatial_metric))}};
+  const auto shift = gr::shift(spacetime_metric, inverse_spatial_metric);
+  const auto lapse = gr::lapse(shift, spacetime_metric);
+  const auto inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+  const auto spacetime_unit_normal_one_form =
+      gr::spacetime_normal_one_form<Dim, Frame::Inertial>(lapse);
+  const auto spacetime_unit_normal_vector =
+      gr::spacetime_normal_vector(lapse, shift);
+
+  const auto dt_spacetime_metric =
+      time_derivative_of_spacetime_metric(lapse, shift, pi, phi);
+  const auto dt_spatial_metric =
+      time_deriv_of_spatial_metric(lapse, shift, phi, pi);
+
+  const auto d_lapse =
+      spatial_deriv_of_lapse(lapse, spacetime_unit_normal_vector, phi);
+  const auto d_shift = spatial_deriv_of_shift(
+      lapse, inverse_spacetime_metric, spacetime_unit_normal_vector, phi);
+  const auto d_spatial_metric = deriv_spatial_metric(phi);
+
+  const auto ex_curv =
+      extrinsic_curvature(spacetime_unit_normal_vector, pi, phi);
+  const auto trace_ex_curv = trace(ex_curv, inverse_spatial_metric);
+  const auto christoffel_first = gr::christoffel_first_kind(d_spatial_metric);
+  const auto trace_christoffel_first =
+      trace_last_indices(christoffel_first, inverse_spatial_metric);
+
+  typename DampedHarmonicCompute<Frame::Inertial>::return_type
+      gauge_h_and_deriv{};
+  DampedHarmonicCompute<Frame::Inertial>::function(
+      make_not_null(&gauge_h_and_deriv), lapse, shift,
+      spacetime_unit_normal_one_form, sqrt_det_spatial_metric,
+      inverse_spatial_metric, spacetime_metric, pi, phi, coords, sigma_r);
+  const auto& gauge_h =
+      get<Tags::GaugeH<Dim, Frame::Inertial>>(gauge_h_and_deriv);
+
+  // Compute lapse and shift time derivatives
+  Scalar<DataVector> dt_lapse{};
+  get(dt_lapse) =
```

Comment:
these calculations are a little surprising to me -- for instance, there's parts that I would have anticipated to be computed equivalently to `time_deriv_of_lapse`. I take it that the parts proportional to `gauge_h` components are additional sources associated with the gauge function, but in each of these calculations there's parts that appear independent of the gauge. Would it be incorrect to calculate those parts from the existing free functions?

Reply:
I would have to go through the math in detail. This is implementing Eq 9 and 10 of https://arxiv.org/pdf/gr-qc/0512093.pdf directly, which I find might be clearer :shrug:

Reply:
Ah, fair enough -- if this is just directly implementing a published result, I won't ask for math alterations. Would you mind adding a citation to that paper in the dox?